EIPS/eip-1948.md

```diff
@@ -0,0 +1,127 @@
+---
+eip: 1948
+title: Non-fungible Data Token
+author: Johann Barbie (@johannbarbie), Ben Bollen <ben@ost.com>, pinkiebell (@pinkiebell)
+discussions-to: https://ethereum-magicians.org/t/erc-non-fungible-data-token/3139
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-04-18
+requires: 721
+---
+
+## Simple Summary
+
+Some NFT use-cases require to have dynamic data associated with a non-fungible token that can change during its lifetime. Examples for dynamic data:
+- cryptokitties that can change color
+- intellectual property tokens that encode rights holders
+- tokens that store data to transport them across chains
+
+The existing metadata standard does not suffice as data can only be set at minting time and not modified later.
+
+## Abstract
+
+Non-fungible tokens (NFTs) are extended with the ability to store dynamic data. A 32 bytes data field is added and a read function allows to access it. The write function allows to update it, if the caller is the owner of the token. An event is emitted every time the data updates and the previous and new value is emitted in it.
+
+## Motivation
+
+The proposal is made to standardize on tokens with dynamic data. Interactions with bridges for side-chains like xDAI or Plasma chains will profit from the ability to use such tokens. Protocols that build on data tokens like [distributed breeding](https://ethresear.ch/t/a-distributed-breeding-function/5264) will be enabled.
+
+## Specification
+
+An extension of [ERC721](https://eips.ethereum.org/EIPS/eip-721) interface with the following functions and events is suggested:
+
+```
+pragma solidity ^0.5.2;
+
+interface IERC1948 {
+  
+  event DataUpdated(uint256 indexed tokenId, bytes32 oldData, bytes32 newData);
+
+  function readData(uint256 _tokenId) public view returns (bytes32);
+
+  function writeData(uint256 _tokenId, bytes32 _newData) public;
+
+}
+```
+
+## Rationale
+
+The suggested data field in the NFT is used either for storing data directly, like a counter or address. If more data is required the implementer should fall back to authenticated data structures, like merkle- or patricia-trees.
+
+The proposal for this ERC stems from the [distributed breeding proposal](https://ethresear.ch/t/a-distributed-breeding-function/5264) to allow better integration of NFTs across side-chains. [ost.com](https://ost.com/), [Skale](https://skalelabs.com/), [POA](https://poa.network/), and [LeapDAO](https://leapdao.org/) have been part of the discussion.
+
+## Backwards Compatibility
+
+🤷‍♂️ No related proposals are known to the author, hence no backwards compatibility to consider.
+
+## Test Cases
+
+Simple happy test:
+
+``` solidity
```

Comment:
This seems to be javascript?

Reply:
Also not sure test case should be as a javascript code and not just the actual data to be sent to the contract and expected return value.

Reply:
is there any specific format this should be in? if not javascript?

Reply:
I haven't seen too many tests included in ERCs to be honest. I think this can be improved at a later time after more discussion took place on the ERC.